### PR TITLE
Implement flex sensor

### PR DESCRIPTION
**Pull request by mck1117**
#540

### CODE REVIEW

File: firmware/controllers/sensors/flex_sensor.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	// this long timeout is ok, ethanol content isn't going to change quickly
+	: StoredValueSensor(SensorType::FuelEthanolPercent, MS2NT(500))
+{
+}
+
+void FlexFuelSensor::onEdge(efitick_t nowNt) {
+	float frequency = 1 / m_edgeTimer.getElapsedSecondsAndReset(nowNt);
+
+	// TODO: are these limits reasonable?
+	if (frequency > 45 && frequency < 155) {
```
Comment (cookiesowns) [inline firmware/controllers/sensors/flex_sensor.cpp:31]: Limits might be a bit broad, most people seem to settle at 49hz and 151hz for floor and ceil.

And signals above 175hz indicate a fault condition ( fuel containmination to trigger failsafe ), we can probably add it as a TODO

Comment (mck1117) [reply firmware/controllers/sensors/flex_sensor.cpp:31]: Good to know. Doesn't really hurt to be a little generous here, as 0.3% is a little tight on the high end (though E100 is also rare).

Comment (cookiesowns) [reply firmware/controllers/sensors/flex_sensor.cpp:31]: Yup, we should be fine with these values 👍 LGTM!

**Comment by mck1117** [comment]
tested and working on real hardware!

**Pull request merged**